components/ContainedInputViews/src/MDCContainedInputView.h

```diff
@@ -0,0 +1,319 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+/**
+ A set of Contained Input View states outlined in the Material guidelines. These states overlap with
+ and extend UIControlState.
+ */
+typedef NS_OPTIONS(NSInteger, MDCContainedInputViewState) {
+  /**
+   The default state of the contained input view.
+   */
+  MDCContainedInputViewStateNormal = 1 << 0,
+  /**
+   The state the view is in during normal editing.
+   */
+  MDCContainedInputViewStateFocused = 1 << 1,
+  /**
+   This state most closely resembles the @c selected UIControlState.
+   */
+  MDCContainedInputViewStateActivated = 1 << 2,
```

Comment:
Clarification:: What is the difference between Focused and Activated?

Reply:
From the [material.io guidelines for text fields](https://material.io/design/components/text-fields.html):

Focused:
<img width="267" alt="Screen Shot 2019-03-25 at 4 05 43 PM" src="https://user-images.githubusercontent.com/8020010/54950583-f84abb00-4f17-11e9-9d79-994b73ae5009.png">
Activated:
<img width="263" alt="Screen Shot 2019-03-25 at 4 05 48 PM" src="https://user-images.githubusercontent.com/8020010/54950582-f84abb00-4f17-11e9-9ba7-0b13b316e9f5.png">

Focused is the state during editing. Activated I think is analogous to UIControlStateSelected, but I'm not sure...

Reply:
It is entirely possible. I don't actually know. We could get rid of it? 

A user would do something like `inputTextField.isActivated = YES;`